core/lib/Cardano/Address/Style/Shelley.hs

```diff
@@ -374,6 +376,19 @@ deriveStakingPrivateKey =
 -- > bech32 $ pointerAddress tag (toXPub <$> addrK) ptr
 -- > "addr1gxpfffuj3zkp5g7ct6h4va89caxx9ayq2gvkyfvww48sdnmmqypqfcp5um"
 
+instance Internal.RewardAccAddress Shelley where
+    rewardAccAddress discrimination k = unsafeMkAddress $
+        invariantSize expectedLength $ BL.toStrict $ runPut $ do
+            putWord8 firstByte
+            putByteString (blake2b224 k)
+      where
+          -- First 4 bits are `1110` for keyhash28 reward account address.
+          -- Next 4 bits are network discriminator.
+          -- `1110 0000` is 224 in decimal.
+          firstByte =
+              224 + invariantNetworkTag 16 (networkTag @Shelley discrimination)
```

Comment:
This is the jist of it... I can verify this "works", but I'm not sure the semantics are correct.

Reply:
@jonathanknowles I know this is a magic number... but I really cannot say what name would be reasonable here.

Reply:
`addrType` would be a reasonable name if you're looking for inspiration.

Reply:
While at it... what's the magic number 16?